### DISCUSSION TITLE
Getting results as 0% for Task1b

### DISCUSSION THREAD

**Discussion by erlik-han** [cafawo/FinancialDataAnalytics#7] [Q&A] [answered]
I have one question abouth my result I am getting this result when I run my code.Anyone has this kind of results
Probability of negative cash position after 6 months: 0.00%
Probability of negative cash position after 12 months: 0.00%
The code is this:

```
# Count the number of simulations where cash position is negative after 6 months
negative_cash_6m = sum(process[6] < 0)

# Estimate the probability of negative cash position after 6 months
prob_negative_cash_6m = negative_cash_6m / len(process[6])

# Count the number of simulations where cash position is negative after 12 months
negative_cash_12m = sum(process[12] < 0)

# Estimate the probability of negative cash position after 12 months
prob_negative_cash_12m = negative_cash_12m / len(process[12])

# Print the results
print("Probability of negative cash position after 6 months: {:.2%}".format(prob_negative_cash_6m))
print("Probability of negative cash position after 12 months: {:.2%}".format(prob_negative_cash_12m))
```

Thanks a lot for your contributions

**Comment by ansenpjames** [comment]
I am also getting 0 in both cases.. don't know whether it is right!!!

**Comment by cafawo** [comment, marked as answer]
Given your code, it seems to me that you simulate one realization of the process for different points (months) in time. However, you need to simulate a large number  of realizations (e.g. sth. like 100k) in order to build a meaningful average. 

Also, you do not really need MC here, you can solve the task analytically (note that the a Wiener process implies a Normal distribution), see: https://en.wikipedia.org/wiki/Normalization_(statistics).

PS: Expect a result slightly above 1% for 12 months.

**Reply by erlik-han** [reply]
I solved the problem thanks I got these results;
Probability of negative cash position in 6 months: 0.0040
Probability of negative cash position in 12 months: 0.0105
